app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/MapsActivity.java

```diff
@@ -82,44 +90,50 @@ protected void onCreate(Bundle savedInstanceState) {
         setUpSearch();
     }
 
-    private void setUpSearch(){
+    private void setUpSearch() {
         mSearchView = (FloatingSearchView) findViewById(R.id.search_view);
         mSearchView.setOnQueryChangeListener(new FloatingSearchView.OnQueryChangeListener() {
             @Override
             public void onSearchTextChanged(String oldQuery, String newQuery) {
-            Map<String, String> map = new HashMap<String, String>();
+                handler.removeCallbacks(workRunnable);
+                workRunnable = () -> autoCompleteRequest(newQuery);
+                handler.postDelayed(workRunnable, 250 /*delay*/);
+            }
+        });
+    }
 
-            map.put("Content-Type", "application/json");
+    private void autoCompleteRequest(String query) {
+        Map<String, String> map = new HashMap<String, String>();
 
-            JSONObject searchJSON = new JSONObject();
-            try {
-                searchJSON.put("query", newQuery);
-            } catch (JSONException e) {
-                e.printStackTrace();
-            }
+        map.put("Content-Type", "application/json");
 
-            final RequestBody requestBody = RequestBody.create(MediaType.get("application/json; charset=utf-8"), searchJSON.toString());
+        JSONObject searchJSON = new JSONObject();
+        try {
+            searchJSON.put("query", query);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
 
-            Endpoint searchEndpoint = new Endpoint()
-                    .path("search")
-                    .body(Optional.of(requestBody))
-                    .headers(map)
-                    .method(Endpoint.Method.POST);
+        final RequestBody requestBody = RequestBody.create(MediaType.get("application/json; charset=utf-8"), searchJSON.toString());
```

Comment:
Might consider breaking this line. It's a bit long.

Reply:
Done!